timApp/velp/velpgroups.py

```diff
@@ -51,6 +49,56 @@ class CreatedVelpGroup:
     created_new_group: bool = True
 
 
+@dataclass
+class GroupSelection:
+    """Stores information on a velp group's selection status"""
+
+    id: int
+    selected: bool
+
+    def to_json(self) -> dict[str, int | bool]:
+        return {"id": self.id, "selected": self.selected}
+
+
+@dataclass
+class VelpGroupSelectionInfo:
+    """Stores information on selected and default velp groups for a document. Used for
+    passing velp group selection data to the user interface.
+    """
+
+    target_ids: list[str]
+    selections: list[list[GroupSelection]]
+
+    def append(self, target_id: str, gs: GroupSelection) -> None:
+        if len(target_id) == 0:
+            return
+        else:
+            index = -1
+            for i in range(len(self.target_ids)):
+                if target_id == self.target_ids[i]:
+                    index = i
+                    break
+            if index < 0:
+                # if the index was not found, we need to create a new entry
+                self.target_ids.append(target_id)
+                self.selections.append([])
+                index = self.target_ids.index(target_id)
+        self.selections[index].append(gs)
+
+    # TODO IVelpGroupCollection in velptypes.ts currently uses Record
+    #      in the same form as below. A simpler format probably requires
+    #      still more refactoring, in the UI code as well.
+    def to_json(self) -> dict[str, list | list[dict[str, int | bool]]]:
```

Comment:
Tässäkin varmaan riittäisi `dict` palautusarvoksi?

```suggestion
    def to_json(self) -> dict:
```

(koska siis tämä on `to_json`)